src/DarkTheme.cpp

```diff
@@ -0,0 +1,51 @@
+/***************************************************************************
+ *   Copyright (C) 2021 by Manuel Wegmann - wegmann.manuel@yahoo.com       *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, write to the                         *
+ *   Free Software Foundation, Inc.,                                       *
+ *   59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.             *
+ ***************************************************************************/
+
+#include "DarkTheme.h"
+#include "AltFocusMenuBarDisable.h"
+
+//DarkTheme only works with Fusion style
```

Comment:
Does selecting the "Dark theme" also force the use of the Fusion style? If so then we are going to have to rethink this as that will produce too many styling changes for installations where another style is used by default (Windows, MacOs, - and Linux/FreeBSD users using the Kvantum style engine which already does dark themes...!)

Reply:
For example, in my system that falls into the last category my current setup gives:
![image](https://user-images.githubusercontent.com/6163092/134828261-e893b3d2-59f6-46fb-849d-df5a7164b0a3.png)

with this option enabled that changes to:
![image](https://user-images.githubusercontent.com/6163092/134828235-20aaa2a2-025b-4a87-91ea-3a30b81c4656.png)

Reply:
I'd use a combobox here and say "Theme: light/dark (built/in)". We can then, in a later PR, extend it with an "auto" option that follows the desktop preferences.

Reply:
TBH I do want there to be a "Dark Desktop theme" option because then I can restore the coloured texts for the different trigger types (and do away with the coloured indicator icons in the item type `QComboBox` selector and the `QLineEdit` for the item data) - and also fix some other elements that don't look good in a Dark DE...

\<aside> ... this was one thing that would work in conjunction with the Application styling improvements I intended for the ~~tossed into a dark corner and left to fester~~ *aborted* #3373 as it would set some aspects of a default "Mudlet" application style sheet - or rather switch between a pair of them, e.g. I had the following lined up:
`mudlet_light.qss`:
```css
/* Version 1.0.0 - this must be the first line as the opening comment marker,
 * space, version word, space and version number are used to track the edition
 * so that a later one will replace a lower number predecessor.
 */

/*
 * The following are used to color the QLineEdit component of the trigger items
 * to restore the color that some were missing - this is for light desktop
 * environments and the colors are somewhat darker (closer to black) than
 * the corresponding items in the one for the dark desktops.
 */

/*
 * This one is not used - it is the default type so we keep the default colors
 * from the DE - which is likely to be a dark color for text on a light
 * background - should a colored hue be needed to match a non-neutral colored
 * skinner's theme than it is recommended to check that the other colors here
 * are visibly different then the value for the default one:
 *
 * .QLineEdit[TriggerPatternItemType="subString"] { color: black; }
 */

 .QLineEdit[TriggerPatternItemType="perlRegex"] { color: blue; }
 .QLineEdit[TriggerPatternItemType="beginningOfLineSubString"] { color: maroon; }
 .QLineEdit[TriggerPatternItemType="exactMatchString"] { color: green; }
 .QLineEdit[TriggerPatternItemType="luaCode"] { color: teal; }
 .QLineEdit[TriggerPatternItemType="lineSpacer"] { color: purple; }

/*
 * These are for the top part of the QTreeView (from which the TTreeWidget is
 * derived) - unfortunately it is not possible to do this for the other
 * TTreeWidgets in the editor at present - because we can only style the header
 * or all the items and Mudlet currently hides the real header and replaces it
 * with the first actual QTreeWidgetItem - and it is not possible to select only
 * a single item by index or other programmable means:
 */
 .QHeaderView[TreeWidgetHeaderType="SearchResults"]::section {
     background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                       stop: 0 #9E9E9E, stop: 0.5 #AFAFAF,
                                       stop: 0.6 #BCBCBC, stop:1 9A9A9A);
     color: black;
     padding-left: 4px;
     border: 1px solid #939393; }
 .QHeaderView[TreeWidgetHeaderType="SearchResults"]::section:checked { background-color: maroon; }
 .QHeaderView[TreeWidgetHeaderType="SearchResults"]::down-arrow { image: url(down_arrow.png); }
 .QHeaderView[TreeWidgetHeaderType="SearchResults"]::up-arrow { image: url(up_arrow.png); }
```
and `mudlet_dark.qss`:
```css
/* Version 1.0.0 - this must be the first line as the opening comment marker,
 * space, version word, space and version number are used to track the edition
 * so that a later one will replace a lower number predecessor.
 */

/*
 * The following are used to color the QLineEdit component of the trigger items
 * to restore the color that some were missing - this is for dark desktop
 * environments and the colors are somewhat lighter (closer to white) than
 * the corresponding items in the one for the light desktops.
 */

/*
 * This one is not used - it is the default type so we keep the default colors
 * from the DE - which is likely to be a light color for text on a dark
 * background - should a colored hue be needed to match a non-neutral colored
 * skinner's theme than it is recommended to check that the other colors here
 * are visibly different then the value for the default one:
 *
 * .QLineEdit[TriggerPatternItemType="subString"] { color: white; }
 */

 .QLineEdit[TriggerPatternItemType="perlRegex"] { color: #8080FF; }
 .QLineEdit[TriggerPatternItemType="beginningOfLineSubString"] { color: red; }
 .QLineEdit[TriggerPatternItemType="exactMatchString"] { color: lime; }
 .QLineEdit[TriggerPatternItemType="luaCode"] { color: aqua; }
 .QLineEdit[TriggerPatternItemType="lineSpacer"] { color: fuchsia; }

/*
 * These are for the top part of the QTreeView (from which the TTreeWidget is
 * derived) - unfortunately it is not possible to do this for the other
 * TTreeWidgets in the editor at present - because we can only style the header
 * or all the items and Mudlet currently hides the real header and replaces it
 * with the first actual QTreeWidgetItem - and it is not possible to select only
 * a single item by index or other programmable means:
 */
 .QHeaderView[TreeWidgetHeaderType="SearchResults"]::section {
     background-color: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                       stop: 0 #616161, stop: 0.5 #505050,
                                       stop: 0.6 #434343, stop:1 #656565);
     color: white;
     padding-left: 4px;
     border: 1px solid #6c6c6c; }
 .QHeaderView[TreeWidgetHeaderType="SearchResults"]::section:checked { background-color: red; }
 .QHeaderView[TreeWidgetHeaderType="SearchResults"]::down-arrow { image: url(down_arrow.png); }
 .QHeaderView[TreeWidgetHeaderType="SearchResults"]::up-arrow { image: url(up_arrow.png); }
```
\</aside>